migrate.ts

```diff
@@ -4,53 +4,76 @@
  * If no environment is specified, you can use the environment variables PG_URL
  * and SETTINGS_FILE
  */
-require("ts-node/register");
-const { getDatabaseConfig, startSshTunnel } = require("./scripts/startup/buildUtil");
-
-const initGlobals = (args, isProd) => {
-  global.bundleIsServer = true;
-  global.bundleIsTest = false;
-  global.bundleIsE2E = false;
-  global.bundleIsProduction = isProd;
-  global.bundleIsMigrations = true;
-  global.defaultSiteAbsoluteUrl = "";
-  global.serverPort = 5001;
-  global.estrellaPid = -1;
+
+import { existsSync } from "node:fs";
+import type { ITask } from "pg-promise";
+
+// @ts-ignore This is a javascript file without a .d.ts
+import { getDatabaseConfig, startSshTunnel } from "./scripts/startup/buildUtil";
+
+const initGlobals = (args: Record<string, unknown>, isProd: boolean) => {
+  Object.assign(global, {
+    bundleIsServer: true,
+    bundleIsTest: false,
+    bundleIsE2E: false,
+    bundleIsProduction: isProd,
+    bundleIsMigrations: true,
+    defaultSiteAbsoluteUrl: "",
+    serverPort: 5001,
+    estrellaPid: -1,
+  });
 
   const { getInstanceSettings } = require("./packages/lesswrong/lib/executionEnvironment");
   getInstanceSettings(args); // These args will be cached for later
 }
 
-const fetchImports = (args, isProd) => {
+const fetchImports = (args: Record<string, unknown>, isProd: boolean) => {
   initGlobals(args, isProd);
 
   const { getSqlClientOrThrow, setSqlClient } = require("./packages/lesswrong/server/sql/sqlClient");
   const { createSqlConnection } = require("./packages/lesswrong/server/sqlConnection");
   return { getSqlClientOrThrow, setSqlClient, createSqlConnection };
 }
 
-const credentialsPath = (forumType) => {
+type ForumType = "lw" | "ea" | "none";
+
+const detectForumType = async (): Promise<ForumType> => {
+  const base = process.env.GITHUB_WORKSPACE ?? "..";
```

Comment:
Regrettably I suspect that not all LW devs have `LessWrong-Credentials` living in the same directory as `ForumMagnum`.  Am not really sure how to solve it in a convenient way... could just have this act as a forcing function, or just make everyone create an empty directory with that name (or a symlink?)

Reply:
Oh, so it is.  For some reason I remembered this being an issue at some other point in time, but maybe I was remembering something else...